WordPress/src/main/java/org/wordpress/android/ui/prefs/timezone/SiteSettingsTimezoneViewModel.kt

```diff
@@ -0,0 +1,211 @@
+package org.wordpress.android.ui.prefs.timezone
+
+import android.app.Activity
+import android.text.TextUtils
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import com.android.volley.Response
+import com.android.volley.VolleyError
+import com.android.volley.toolbox.StringRequest
+import com.android.volley.toolbox.Volley
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import org.wordpress.android.Constants
+import org.wordpress.android.networking.RestClientUtils
+import org.wordpress.android.ui.prefs.timezone.TimezonesList.TimezoneHeader
+import org.wordpress.android.ui.prefs.timezone.TimezonesList.TimezoneItem
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T.SETTINGS
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.time.format.TextStyle.FULL
+import java.time.zone.ZoneRulesException
+import java.util.Locale
+import javax.inject.Inject
+
+class SiteSettingsTimezoneViewModel @Inject constructor() : ViewModel() {
+    private val timezonesList = mutableListOf<TimezonesList>()
+
+    private val _showEmpty = MutableLiveData<Boolean>()
+    val showEmptyView: LiveData<Boolean> = _showEmpty
+
+    private val _showProgress = MutableLiveData<Boolean>()
+    val showProgressView: LiveData<Boolean> = _showProgress
+
+    private val _dismiss = MutableLiveData<Unit>()
+    val dismissWithError: LiveData<Unit> = _dismiss
+
+    private val searchInput = MutableLiveData<String>()
+    val timezoneSearch: LiveData<List<TimezonesList>> = Transformations.switchMap(searchInput) { term ->
+        filterTimezones(term)
+    }
+
+    private val _suggestedTimezone = MutableLiveData<String>()
+    val suggestedTimezone = _suggestedTimezone
+
+    private val _timezones = MutableLiveData<List<TimezonesList>>()
+    val timezones = _timezones
+
+
+    fun searchTimezones(query: CharSequence?) {
+        searchInput.value = query.toString()
```

Comment:
`query` is marked as nullable, so if at any point it will actually be a null, we will filter with a "null" string, which might not be what want :)